Add S3 Sync and ENV file

john-jac opened the pull request:
*Issue #, if available:*

*Description of changes:*

Allows local-runner to sync dags, requirements, and plugins.zip from S3 just as MWAA does.  Also adds an environment file in the same way as https://github.com/aws/aws-mwaa-local-runner/pull/95

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

john-jac committed:
Add S3 Sync and ENV file

Kalhan22 commented on docker/script/entrypoint.sh:
```diff
@@ -81,14 +81,35 @@ fi
+      echo "Syncing $S3_DAGS_PATH"   
+      mkdir -p /usr/local/airflow/dags
+      cd /usr/local/airflow/dags
+      aws s3 sync --exact-timestamp $S3_DAGS_PATH .
```
Adding a --delete flag to the command might be useful. Ensures un necessary or stale files are deleted during sync. In the absence of the flag the user might happen to have differences between the local and cloud locations and not notice them.

https://docs.aws.amazon.com/cli/latest/reference/s3/sync.html

ashishgo-aws commented on docker/script/entrypoint.sh:
```diff
@@ -81,14 +81,35 @@ fi
+    # if S3_REQUIREMENTS_PATH
+    if [ -n "$S3_REQUIREMENTS_PATH" ]; then
+      echo "Downloading $S3_REQUIREMENTS_PATH"
+      aws s3 cp $S3_REQUIREMENTS_PATH /usr/local/airflow/dags/requirements.txt
```
If for some reason, `S3_DAGS_PATH` is missing, then the folder `/usr/local/airflow/dags` will not be present. 

Maybe there should be an error if `S3_DAGS_PATH` is not provided or we can place the requirements file in its own folder.

john-jac committed:
Moved requirements into own folder

This matches what MWAA does, and also allows adding --delete into S3 sync command.  Also fixed pip3 install error.
